pkg/certificate/types.go

```diff
@@ -125,6 +128,7 @@ type Manager struct {
 type MRCClient interface {
 	List() ([]*v1alpha2.MeshRootCertificate, error)
 	MRCEventBroker
+	UpdateMeshRootCertificate(mrc *v1alpha2.MeshRootCertificate) error
```

Comment:
Could we embed the `compute.Interface` here? We could avoid adding the `UpdateMeshRootCertificate` method signature and also have access to other MRC update/get/list methods

Reply:
Could you explain for my own understanding why we're not adding the `compute.Interface` here? Thanks!

Reply:
it would pull in all the other unused methods. interfaces should be thin, and describe only what is necessary. It makes creating another implementation much easier, and creates a clear definition of exactly what is needed.

Take a look at some of the goals and guiding principles in the refactoring doc https://docs.google.com/document/d/1L_u-XlVU8Fog-uFS3Uf0O2nz8IArQy82cd6Npxm038Y/edit#heading=h.cvqx4tbzt6i7

Reply:
Okay, I must have misunderstood this comment on my PR https://github.com/openservicemesh/osm/pull/5073#discussion_r970050870